File: ITapply/ITapply.Models/SearchObjects/EducationSearchObject.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ITapply.Models.SearchObjects
{
    public class EducationSearchObject : BaseSearchObject
    {
        public int? CandidateId { get; set; }
        public string Institution { get; set; } = string.Empty;
        public string Degree { get; set; } = string.Empty;
        public string FieldOfStudy { get; set; } = string.Empty;
        public bool? IsCurrent { get; set; }
        public DateTime? StartDateFrom { get; set; }
        public DateTime? StartDateTo { get; set; }
        public DateTime? EndDateFrom { get; set; }
        public DateTime? EndDateTo { get; set; }
    }
} 

File: ITapply/ITapply.UI/itapply_desktop/lib/models/search_objects/education_search_object.dart
import 'package:itapply_desktop/models/search_objects/base_search_object.dart';
import 'package:json_annotation/json_annotation.dart';

part 'education_search_object.g.dart';

@JsonSerializable()
class EducationSearchObject extends BaseSearchObject {
  int? CandidateId;
  String? Institution;
  String? Degree;
  String? FieldOfStudy;
  bool? IsCurrent;
  DateTime? StartDateFrom;
  DateTime? StartDateTo;
  DateTime? EndDateFrom;
  DateTime? EndDateTo;

  EducationSearchObject({
    this.CandidateId,
    this.Institution,
    this.Degree,
    this.FieldOfStudy,
    this.IsCurrent,
    this.StartDateFrom,
    this.StartDateTo,
    this.EndDateFrom,
    this.EndDateTo,
    super.Page,
    super.PageSize,
    super.IncludeTotalCount,
    super.RetrieveAll,
  });

  @override
  Map<String, dynamic> toJson() => _$EducationSearchObjectToJson(this);
}